include/pmacc/particles/memory/dataTypes/Particle.hpp

```diff
@@ -139,9 +137,7 @@ namespace pmacc
 
         /** const version of method operator(const T_Key) */
         template<typename T_Key>
-        HDINLINE typename boost::result_of<
-            typename std::remove_reference_t<typename boost::result_of<const FrameType(T_Key)>::type>(uint32_t)>::type
-        operator[](const T_Key key) const
+        HDINLINE const auto& operator[](const T_Key key) const
```

Comment:
I'm not sure, but these operators look like forwarding, should those return `decltype(auto)` as well?

Reply:
I decided to stick with explicitely stating that the return value is a const reference. The return value refers to 1 element of the array<T, 256> for `T_Key` held by the frame. I thought it gives a better idea of what this return type is. In fact, on my llama branch, I changed this into `decltype(auto)` because then this method can also return proxy references.

Reply:
Okay, thanks for explaining.